src/sql/azdata.proposed.d.ts

```diff
@@ -847,7 +847,7 @@ declare module 'azdata' {
 		 * @param tableInfo The table information. The object will be passed back to the table designer provider as the unique identifier for the table.
 		 * @param telemetryInfo Optional Key-value pair containing any extra information that needs to be sent via telemetry
 		 */
-		export function openTableDesigner(providerId: string, tableInfo: TableInfo, telemetryInfo?: { [key: string]: string }): Thenable<void>;
+		export function openTableDesigner(providerId: string, tableInfo: TableInfo, telemetryInfo?: { [key: string]: string }, objectExplorerContext?: ObjectExplorerContext): Thenable<void>;
```

Comment:
Add param doc comment

Is there a reason this is optional? If that's something you specifically want to support make sure you're handling it not being available correctly below (see my other comment)

Reply:
done
This being optional is by design, sqlproj also supports table designer but doesn't have OE context. I've added extra check for the context not being available and tested with sqlproj designer publishing and it's working fine.